lib/chef/userable.rb

```diff
@@ -0,0 +1,106 @@
+#
+# Author:: Snehal Dwivedi (sdwivedi@chef.io)
+# Copyright:: Copyright (c) 2008-2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# common methods of User and UserV1
+class Chef
+  class Userable
```

Comment:
This should probably be something like Chef::Mixin::UserBase or something, although I don't like the Mixin directory naming scheme, but I can't think of anything better offhand right now.

Reply:
@phiggins might have some better ideas than i do

Reply:
I agree, I'd prefer something like `AbstractUser` or `BaseUser` to `Userable`.